auth/validator.py

```diff
@@ -0,0 +1,52 @@
+from typing import Dict
+
+import httpx
+from fastapi import HTTPException
+from jose import jwt
+from jose.exceptions import JWTError
+
+from auth.config import get_settings
+
+
+def verify_jwt(token: str) -> Dict:
+    settings = get_settings()
+    try:
+        jwks_url = f"https://{settings.auth0_domain}/.well-known/jwks.json"
+        response = httpx.get(jwks_url)
+        jwks = response.json()
+
+        unverified_header = jwt.get_unverified_header(token)
+        rsa_key = {}
```

Comment:
If you're going to use an empty value for a value that might not be found/set, I'd recommend `None`. I think the process of trying to find a matching key for the token should be split off into a function like:

```python
def get_rsa_key(token: str) -> dict | None:
```

And then to check if it's `None` do something like:

```python
rsa_key = get_rsa_key(token)

if rsa_key is None:
    raise HTTPException(...)
```

rather than having all the rest of the logic under an `if` check